pkg/platform/kube/monitoring/function.go

```diff
@@ -164,29 +164,42 @@ func (fm *FunctionMonitor) updateFunctionStatus(function *nuclioio.NuclioFunctio
 	return nil
 }
 
-// consider function deployment available if it has at least one available pod
+// consider function deployment available when all pods are available
 func (fm *FunctionMonitor) isAvailable(deployment *appsv1.Deployment) bool {
-	return !(deployment.Spec.Replicas != nil &&
+	atLeastOneRunningPod := !(deployment.Spec.Replicas != nil &&
```

Comment:
Not accurate since if replicas=0  it will still be true.
suggest to restructure this entire method in a clearer way:
```
atLeastOneReplicasRequsted := !(deployment.Spec.Replicas != nil && *deployment.Spec.Replicas > 0)
return atLeastOneReplicasRequsted || deployment.Status.UnavailableReplicas == 0
```

Reply:
im not sure what you suggeted fit in the desired question here
we want to ensure that
1. there is at least one available replica
2. no unavailable replicas

perhaps we should
```
return deployment.Status.UnavailableReplicas == 0 && deployment.Spec.Replicas != nil && *deployment.Spec.Replicas > 0
```

thoughts?

Reply:
1 is not accurate since function can be scaled to zero or configured with 0 replicas

Reply:
You made up a good point and I found out that "disabling" a function is bugged due because resync wasn't handling "ready" state anymore. Fixed that "on the way" + added test.

I used 

```golang
atLeastOneReplicasRequested := deployment.Spec.Replicas != nil && *deployment.Spec.Replicas > 0
return atLeastOneReplicasRequested && deployment.Status.UnavailableReplicas == 0
``` 

to ensure (1) one replicas is requested (2) no unavailable replicas